Detector.cc

```
@@ -1,6308 +1,6760 @@
-            stations[StationIndex].strings[stringNum].antennas[antennaNum].SetY(stations[StationIndex].GetY()+araGeom->getStationInfo(StationID)->fAntInfo[antId].antLocation[1]);
-            //stations[stationNum].strings[stringNum].antennas[antennaNum].SetZ(araGeom->fStationInfo[stationNum].fAntInfo[chan-1].antLocation[2]-double(settings1->DEPTH_CHANGE));
-            stations[StationIndex].strings[stringNum].antennas[antennaNum].SetZ(araGeom->getStationInfo(StationID)->fAntInfo[antId].antLocation[2]);
-            
-            /*
-             cout <<
-             "DetectorStation:string:antenna:X:Y:Z:: " <<
-             i<< " : " <<
-             j<< " : " <<
-             k<< " : " <<
-             stations[i].strings[j].antennas[k].GetX() << " : " <<
-             stations[i].strings[j].antennas[k].GetY() << " : " <<
-             stations[i].strings[j].antennas[k].GetZ() << " : " <<
-             endl;
-             */
-            
-            stations[StationIndex].strings[stringNum].antennas[antennaNum].type = int(araGeom->getStationInfo(StationID)->fAntInfo[antId].polType);  //set polarization to match the deployed information
-            
-            stations[StationIndex].strings[stringNum].SetX(stations[StationIndex].GetX()+araGeom->getStationInfo(StationID)->fAntInfo[antId].antLocation[0]);
-            stations[StationIndex].strings[stringNum].SetY(stations[StationIndex].GetY()+araGeom->getStationInfo(StationID)->fAntInfo[antId].antLocation[1]);
-            
-            
-            
-            if ( params.antenna_orientation == 0 ) {    // all borehole antennas facing same x
-                stations[StationIndex].strings[stringNum].antennas[antennaNum].orient = 0;
-            }
-            else if ( params.antenna_orientation == 1 ) {   // borehole antennas one next facing different way
-                if ( stringNum==0||stringNum==3 ) {
-                    if ( antennaNum==0||antennaNum==1 ) {
-                        stations[Sta
```

What happens in this case? Will the code keep running. Maybe we should throw an error? I would also print this message with `cerr` rather than `cout`

---

I'm guessing you're talking about this line and related DETECTOR==5 part of the function? https://github.com/abigailbishop/AraSim/blob/742170fa7422b3c13ea4c6e6d0c0f25ffb4c6559/Detector.cc#L6264
This function could use a freshening up because I think each DETECTOR=__ part is just for setting `stationId`. So what was going wrong originally is that `stationId` wasn't being set properly for DETECTOR=5 meaning the wrong channel numbers would be returned (and usually it would run for a bit before breaking) or the script wouldn't find a channel number and would return `-1` which makes the code seg fault quickly.

---

(Sorry maybe these comments only show the context properly when you look at the full file diffs)

This is referring to line 1869 (new), specifically the "else" case where the input station is invalid.

---

Yeah if this block of code is reached, AraSim won't run properly. So I did change this from `cout` to `cerr`. Is that all that needs to happen for an error to be thrown?

---

To throw an error (and cause things to crash in a controlled way) you should use `throw runtime_error("Useful error message");`

---

Great thanks. Done